x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameIndexerTransformStats.java

```diff
@@ -133,7 +132,6 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 throw new IllegalArgumentException("when storing transform statistics, a valid transform id must be provided");
             }
             builder.field(DataFrameField.ID.getPreferredName(), transformId);
```

Comment:
this is a good point, the purpose of the ID was to delete corresponding documents after the deletion of a transform (via dbq). As stats is not stored as such the ID is as superfluous as the doc type, so the whole internal storage handling is obsolete.

(We can also clean up in a later PR if preferred, this is already getting complex)

Reply:
I will do this in a followup PR.  It will also mean the assertion that IDs are identical on merge will have to go, but I guess that was only a debugging aid - it didn't do anything in production.